components/draggable/index.js

```diff
@@ -0,0 +1,157 @@
+		this.cloneWrapper.style.left =
+			`${ parseInt( this.cloneWrapper.style.left, 10 ) + event.clientX - this.cursorLeft }px`;
+
+		// Update cursor coordinates.
+		this.cursorLeft = event.clientX;
+		this.cursorTop = event.clientY;
+	}
+
+	/**
+	 *  - Clones the current element and spawns clone over original element.
+	 *  - Adds a fake temporary drag image to avoid browser defaults.
+	 *  - Sets transfer data.
+	 *  - Adds dragover listener.
+	 * @param  {Object} event					The non-custom DragEvent.
+	 * @param  {string} elementId				The HTML id of the element to be dragged.
+	 * @param  {Object} transferData			The data to be set to the event's dataTransfer - to be accessible in any later drop logic.
+	 */
+	onDragStart( event ) {
+		const { elementId, transferData, onDragStart = noop } = this.props;
+		const element = document.getElementById( elementId );
+
+		// Set a fake drag image to avoid browser defaults. Remove from DOM right after.
+		// event.dataTransfer.setDragImage is not supported yet in IE,
+		// we need to check for its existance first.
+		if ( 'function' === typeof event.dataTransfer.setDragImage ) {
+			const dragImage = document.createElement( 'div' );
+			dragImage.id = `drag-image-${ elementId }`;
+			dragImage.classList.add( dragImageClass );
+			document.body.appendChild( dragImage );
+			event.dataTransfer.setDragImage( dragImage, 0, 0 );
+			this.props.setTimeout( () => {
+				document.body.removeChild( dragImage );
+			} );
+		}
+
+		event.dataTransfer.setData( 'text', JSON.stringify( transferData ) );
+
+		// Prepare element clone and append to element wrapper.
+		const elementRect = element.getBoundingClientRect();
+		const elementWrapper = element.parentNode;
+		const elementTopOffset = parseInt( elementRect.top, 10 );
+		const elementLeftOffset = parseInt( elementRect.left, 10 );
+		const clone = element.cloneNode( true );
+		clone.id = `clone-${ elementId }`;
+		this.cloneWrapper = document.createElement( 'div' );
+		this.cloneWrapper.classList.add( cloneWrapperClass );
+		this.cloneWrapper.style.width = `${ elementRect.width + ( clonePadding * 2 ) }px`;
+
+		if ( elementRect.height > cloneHeightTransformationBreakpoint ) {
+			// Scale down clone if original element is larger than 700px.
+			this.cloneWrapper.style.transform = 'scale(0.5)';
+			this.cloneWrapper.style.transformOrigin = 'top left';
+			// Position clone near the cursor.
+			this.cloneWrapper.style.top = `${ event.clientY - 100 }px`;
+			this.cloneWrapper.style.left = `${ event.clientX }px`;
+		} else {
+			// Position clone right over the original element (20px padding).
+			this.cloneWrapper.style.top = `${ elementTopOffset - clonePadding }px`;
+			this.cloneWrapper.style.left = `${ elementLeftOffset - clonePadding }px`;
+		}
+
+		// Hack: Remove iFrames as it's causing the embeds drag clone to freeze
+		clone.querySelectorAll( 'iframe' ).forEach( child => child.remove() );
```

Comment:
`NodeList#forEach` isn't supported in IE11:

https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach#Browser_Compatibility

Reply:
Yes, the cloned node is for the drag image.

I actually tried `dom-to-image` and it failed (JS error in the library regarding SVG elements, maybe they are unsupported). I admit I didn't dig deeper. 

I think the cloned node is a good approach if we want to keep a visual representation but I wonder if we should just provide a dummy Drag Image instead, maybe a white div with a dashed border or something.

Reply:
Hey - not sure if I'm following up to the conversation, but the clone node was used for:

1. Create a drag image that works in IE, as it doesn't support the `setDragImage` call.
2. Have a fully customisable drag image. For this to be aligned with the mocks I needed to control the opacity and shadow of the drag preview, which is not possible via any other route I investigated (I believe the scaling was also an issue). 

p.s. It was possible to clone the node, add the shadow, set it as drag image, discard the clone right away, and let the browser handle the dragging -- simpler, as you don't have to track mouse movement and manually move the clone around. Like I mentioned though, that didn't work with IE, and it was also not possible to control the opacity (the browser would eventually always render the drag image with less opacity). So cloning the node and moving it around manually was the best I could think of.

Reply:
p.s. another source for inspiration going forward: https://github.com/atlassian/react-beautiful-dnd